src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/TransactionBundleValidator.cs

```diff
@@ -0,0 +1,161 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Conformance;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Resources;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Messages.Search;
+using Microsoft.Health.Fhir.Core.Models;
+using static Hl7.Fhir.Model.Bundle;
+
+namespace Microsoft.Health.Fhir.Api.Features.Resources.Bundle
+{
+    public class TransactionBundleValidator : BaseConditionalHandler
+    {
+        public TransactionBundleValidator(
+            IFhirDataStore fhirDataStore,
+            Lazy<IConformanceProvider> conformanceProvider,
+            IResourceWrapperFactory resourceWrapperFactory,
+            ISearchService searchService,
+            ResourceIdProvider resourceIdProvider)
+            : base(fhirDataStore, searchService, conformanceProvider, resourceWrapperFactory, resourceIdProvider)
+        {
+        }
+
+        /// <summary>
+        /// This method validates if transaction bundle contains multiple entries that are modifying the same resource.
+        /// It also validates if the request operations within a entry is a valid operation.
+        /// </summary>
+        /// <param name="bundle"> The input bundle</param>
+        /// <param name="cancellationToken"> The cancellation token</param>
+        public async Task ValidateBundle(Hl7.Fhir.Model.Bundle bundle, CancellationToken cancellationToken)
+        {
+            var resourceIdList = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var entry in bundle.Entry)
+            {
+                if (ShouldValidateBundleEntry(entry))
+                {
+                    string resourceId = await GetResourceId(entry, cancellationToken);
+                    string conditionalCreateQuery = entry.Request.IfNoneExist ?? entry.Request.IfNoneExist;
```

Comment:
This looks redundant?
```suggestion
                    string conditionalCreateQuery = entry.Request.IfNoneExist;
```

Reply:
I was assigning the value only if the conditionalParameter existed. But since I am checking null when I am using the variable. It is redundant check and can be removed. I will remove it.